tf_files/aws/modules/account-management-logs/cloud.tf

```diff
@@ -0,0 +1,175 @@
+            "Action": "s3:GetBucketAcl",
+            "Resource": "arn:aws:s3:::${var.account_name}-management-logs"
+        },
+        {
+            "Sid": "AWSCloudTrailWrite20150319",
+            "Effect": "Allow",
+            "Principal": {
+                "Service": "cloudtrail.amazonaws.com"
+            },
+            "Action": "*",
+            "Resource": "arn:aws:s3:::${var.account_name}-management-logs/*",
+            "Condition": {
+                "StringEquals": {
+                    "s3:x-amz-acl": "bucket-owner-full-control"
+                }
+            }
+        }
+    ]
+}
+POLICY
+}
+
+
+resource "aws_s3_bucket_policy" "b" {
+  bucket = "${aws_s3_bucket.management-logs_bucket.id}"
+
+  policy = "${local.bucket_policy}"
+}
+
+
+# We also need a CloudWatchLogGroup so we can hook it up with the CSOC account through a subscription Filter
+
+resource "aws_cloudwatch_log_group" "management-logs_group" {
+  name = "${var.account_name}_management-logs"
+  tags {
+    Environment = "ALL"
+    Organization = "CTDS"
+  }
+  retention_in_days = 1827
+}
+
+resource "aws_iam_role" "cloudtrail_role" {
+  name = "management-logs_cloudtrail_role"
+  path = "/"
+
+  assume_role_policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Sid": "",
+      "Effect": "Allow",
+      "Principal": {
+        "Service": "cloudtrail.amazonaws.com"
+      },
+      "Action": "sts:AssumeRole"
+    }
+  ]
+}
+EOF
+}
+
+data "aws_iam_policy_document" "cloudtrail_to_cloudwatch_policy_document" {
+  statement {
+    actions = [
+      "logs:*",
+#      "logs:CreateLogsStream",
+#      "logs:PutLogEvents",
+    ]
+    effect = "Allow"
+    resources = [
+      "${aws_cloudwatch_log_group.management-logs_group.arn}:*"
+    ]
+  }
+}
+
+
+resource "aws_iam_role_policy" "cloudtrail_to_cloudwatch_policy" {
+  name   = "${var.account_name}_management-logs_policy"
+  policy = "${data.aws_iam_policy_document.cloudtrail_to_cloudwatch_policy_document.json}"
+  role   = "${aws_iam_role.cloudtrail_role.id}"
+}
+
+
+
+# Create the trail
+
+resource "aws_cloudtrail" "logs-trail" {
+
+  name                        = "${var.account_name}_management_trail"
+  is_multi_region_trail       = true
+  cloud_watch_logs_role_arn   = "${aws_iam_role.cloudtrail_role.arn}"
+  cloud_watch_logs_group_arn  = "${aws_cloudwatch_log_group.management-logs_group.arn}"
+  event_selector {
+    read_write_type           = "All"
+    include_management_events = true
+  }
+
+  s3_bucket_name = "${aws_s3_bucket.management-logs_bucket.id}"
+  s3_key_prefix = "management-logs"
+
+  tags {
+    Environment = "${var.account_name}"
+    Organization = "CTDS"
+  }
+}
+
+
+# Finally the subscription filter
+
+resource "aws_cloudwatch_log_subscription_filter" "csoc_subscription" {
+  name            = "${var.account_name}_subscription"
+  destination_arn = "arn:aws:logs:${data.aws_region.current.name}:${var.csoc_account_id}:destination:management-logs_logs_destination"
```

Comment:
Why not make the entire `destination_arn` a variable rather than require that the destination account follow a naming convention?
Also - can we make the subscription_filter optional? `count = ...`